packages/jest-diff/src/diffStrings.js

```diff
@@ -68,23 +76,24 @@ const diffLines = (a: string, b: string): Diff => {
   };
 };
 
+const showPatchMarks = (hunk: Hunk, a: string): boolean => {
+  const oldLines: Array<string> = a.match(/\n/g) || [];
+  return oldLines.length > hunk.oldLines;
+};
+
 const structuredPatch = (a: string, b: string): Diff => {
   const options = {context: DIFF_CONTEXT};
   let isDifferent = false;
   // Make sure the strings end with a newline.
-  if (!a.endsWith('\n')) {
-    a += '\n';
-  }
-  if (!b.endsWith('\n')) {
-    b += '\n';
-  }
+  a = a.endsWith('\n') ? a : a + '\n';
+  b = b.endsWith('\n') ? b : b + '\n';
```

Comment:
meh, generally I try to avoid situations like this where the common case is a no-op. With the branch before it was more code but would only do an assignment if it needs to, rather than always.